From: Ivan Molodetskikh <yalterz@gmail.com>
Date: Sat, 18 Mar 2023 21:45:41 -0700
Subject: screenshot: Get cursor texture from window

When a window is in the background and should not have the cursor on top
of it, its _cursor will be null. By getting the texture through it, we
add this extra check, which was missing before, leading to a cursor
drawn at 0, 0 on windows where it should not have been drawn.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2702>

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/237c3e9
---
 js/ui/screenshot.js | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/js/ui/screenshot.js b/js/ui/screenshot.js
index 59c63c7..ecbe50b 100644
--- a/js/ui/screenshot.js
+++ b/js/ui/screenshot.js
@@ -917,6 +917,10 @@ class UIWindowSelectorWindow extends St.Button {
         this.insert_child_below(this._cursor, this._border);
     }
 
+    getCursorTexture() {
+        return this._cursor?.content;
+    }
+
     setCursorVisible(visible) {
         if (!this._cursor)
             return;
@@ -1812,7 +1816,7 @@ var ScreenshotUI = GObject.registerClass({
 
             const texture = content.get_texture();
 
-            let cursorTexture = this._cursor.content?.get_texture();
+            let cursorTexture = window.getCursorTexture()?.get_texture();
             if (!this._cursor.visible)
                 cursorTexture = null;
 
